src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -875,233 +873,192 @@ public void setFlags(final int value) {
         setIndexingBin(null);
     }
 
-    /**
-     * the read is paired in sequencing, no matter whether it is mapped in a pair.
-     */
-    public boolean getReadPairedFlag() {
-        return (mFlags & SAMFlag.READ_PAIRED.flag) != 0;
-    }
+    ////////////////////////////////////////////////////////////////////////////
+    // Flag accessor and mutators
+    ////////////////////////////////////////////////////////////////////////////
 
-    private void requireReadPaired() {
-        if (!getReadPairedFlag()) {
-            throw new IllegalStateException("Inappropriate call if not paired read");
-        }
-    }
+    /** Returns true if the "read paired in sequencing" flag is true. */
+    public boolean isPaired() { return getFlag(SAMFlag.READ_PAIRED); }
 
-    /**
-     * the read is mapped in a proper pair (depends on the protocol, normally inferred during alignment).
-     */
-    public boolean getProperPairFlag() {
-        requireReadPaired();
-        return getProperPairFlagUnchecked();
-    }
+    /** Sets the "read paired in sequencing" flag to the given value. */
```

Comment:
I've taken a pass through and tried to be more consistent.  Am happy to take suggestions for anywhere you think could still be improved.